metaflow/linters/ruff_wrapper.py

```diff
@@ -0,0 +1,43 @@
+import os
+import subprocess
+
+from metaflow.cmd.configure_cmd import echo
+from metaflow.exception import MetaflowException
+from metaflow.linters.linter_interface import LinterWrapperInterface
+
+try:
+    from ruff.__main__ import find_ruff_bin
+except ImportError:
+    echo("Package `ruff` is not installed. Please install it before proceeding")
+except Exception as e:
+    raise MetaflowException(e)
+
+
+from metaflow.cmd.configure_cmd import echo
+from metaflow.linters.lint import LintWarn, FlowLinter
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    echo("unable to import StringIO, possibly not installed. Resorting to `io`", fg="yellow")
+    from io import StringIO
+
+from metaflow.exception import MetaflowException
+from metaflow.extension_support import get_aliased_modules
+
+
+class RuffLintWarn(LintWarn):
+    headline = "Ruff is not happy"
+
+
+class RuffWrapper(LinterWrapperInterface):
+    def __init__(self):
+        super().__init__()
+        try:
+            self._binary_path = os.fsdecode(find_ruff_bin())
+        except Exception as e:
+            raise MetaflowException(e)
+
+    def lint(self, file_path: str, warnings=None, lint_config=None, logger=None):
```

Comment:
to be implemented and vetted @tuulos 
the idea is to "subprocess" it as has been also shown in the [ruff.__main__](https://github.com/astral-sh/ruff/blob/main/python/ruff/__main__.py)